### PR TITLE
fix: Change 'rds' to 'sql' in public-facing symbols

### PR DESCRIPTION
**Pull request by palpatim**
#### Description of changes

This change leaves most of the internal symbols unchanged, but should pick up the public-facing ones. Notably:

* Imported schema file name
* Stack name
* SQL related CDK resource IDs
* Some error & warning messages

This PR does NOT override ‘RDS’ in resolver template output, since that would bloat the size of the PR and those files aren't generally reviewed by customers anyway.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/graphql-transformer-common/src/ResourceConstants.ts [hunk 1]
```diff
@@ -29,16 +29,16 @@ export class ResourceConstants {
...
 
-    // RDS
-    RDSLambdaIAMRoleLogicalID: 'RDSLambdaIAMRole',
-    RDSLambdaLogAccessPolicy: 'RDSLambdaLogAccessPolicy',
-    RDSPatchingLambdaIAMRoleLogicalID: 'RDSPatchingLambdaIAMRole',
-    RDSLambdaLogicalID: 'RDSLambdaLogicalID',
-    RDSPatchingLambdaLogAccessPolicy: 'RDSPatchingLambdaLogAccessPolicy',
-    RDSPatchingLambdaLogicalID: 'RDSPatchingLambdaLogicalID',
-    RDSLambdaDataSourceLogicalID: 'RDSLambdaDataSource',
-    RDSLambdaDataSourceLogicalName: 'RDSLambdaDatabase',
```
Comment (palpatim) [inline packages/graphql-transformer-common/src/ResourceConstants.ts:40]: Note: I removed this constant -- it was not in use anywhere and was easily confused with `RDSLambdaDataSourceLogicalID`.

File: packages/graphql-transformer-common/src/ResourceConstants.ts [hunk 2]
```diff
@@ -29,16 +29,16 @@ export class ResourceConstants {
...
-    RDSPatchingSubscriptionLogicalID: 'RDSPatchingSubscriptionLogicalID',
+    // SQL
+    SQLLambdaIAMRoleLogicalID: 'SQLLambdaIAMRole',
+    SQLLambdaLogAccessPolicy: 'SQLLambdaLogAccessPolicy',
+    SQLPatchingLambdaIAMRoleLogicalID: 'SQLPatchingLambdaIAMRole',
+    SQLLambdaLogicalID: 'SQLLambdaLogicalID',
+    SQLPatchingLambdaLogAccessPolicy: 'SQLPatchingLambdaLogAccessPolicy',
+    SQLPatchingLambdaLogicalID: 'SQLPatchingLambda',
+    SQLLambdaDataSourceLogicalID: 'SQLLambdaDataSource',
+    SQLPatchingSubscriptionLogicalID: 'SQLPatchingSubscription',
```
Comment (palpatim) [inline packages/graphql-transformer-common/src/ResourceConstants.ts:40]: Note that I added `SQLPatchingSubscriptionLogicalID`, which was previously hardcoded in the subscription creation logic

**Review by palpatim — COMMENTED** [review]
marker review

File: packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts [hunk 3]
```diff
@@ -58,77 +58,77 @@ const getLatestLayers = (latestLayers?: RDSLayerMapping): RDSLayerMapping => {
   return defaultLayerMapping;
 };
 
-// For beta use account '956468067974', layer name 'AmplifyRDSLayerBeta' and layer version '12' as of 2023-06-20
-// For prod use account '582037449441', layer name 'AmplifyRDSLayer' and layer version '3' as of 2023-06-20
+// For beta use account '956468067974', layer name 'AmplifySQLLayerBeta' and layer version '12' as of 2023-06-20
+// For prod use account '582037449441', layer name 'AmplifySQLLayer' and layer version '3' as of 2023-06-20
```
Comment (alharris-at) [inline packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts:62]: Not a blocker, but comment is out of date.

**Review by alharris-at — DISMISSED** [review]

**Review by dpilch — APPROVED** [review]

**Pull request merged**